parser_test.go

```diff
@@ -6341,6 +6341,32 @@ func (s *testParserSuite) TestPlanRecreator(c *C) {
 	c.Assert(v.Analyze, IsTrue)
 }
 
+func (s *testParserSuite) TestCharsetIntroducer(c *C) {
+	p := parser.New()
+	// `_gbk` is treated as an identifier.
+	_, _, err := p.Parse("select _gbk 'a';", "", "")
```

Comment:
does this mean if someone upgrades its tidb, SQL like `select _gbk 'a'` will report syntax error？it looks like a compatibility breaker. We need to document this change.

Reply:
I think it should be fine because it reports "unknown column" in the old version TiDB.

```sql
mysql> select _gbk 'a';
ERROR 1054 (42S22): Unknown column '_gbk' in 'field list'
mysql> select version();
+--------------------+
| version()          |
+--------------------+
| 5.7.25-TiDB-v5.1.1 |
+--------------------+
1 row in set (0.00 sec)
```

Reply:
right, the `_gbk` is interpreted as an identifier, if someone uses it as the column name, it will be wrong

Reply:
`column_name 'string_literal'` isn't a valid expression anyway.